src/renderer/services/midgard/pools.ts

```diff
@@ -59,71 +59,136 @@ const createPoolsService = (
   getMidgardDefaultApi: (basePath: string) => DefaultApi,
   selectedPoolAsset$: Rx.Observable<O.Option<Asset>>
 ): PoolsService => {
+  const midgardDefailtApi$ = FP.pipe(byzantine$, liveData.map(getMidgardDefaultApi), RxOp.shareReplay(1))
+
   // Factory to get `Pools` from Midgard
   const apiGetPools$ = (request: GetPoolsRequest) =>
-    byzantine$.pipe(
-      liveData.map(getMidgardDefaultApi),
+    FP.pipe(
+      Rx.combineLatest([midgardDefailtApi$, reloadPools$]),
+      RxOp.map(([api]) => api),
       liveData.chain((api) =>
         FP.pipe(
           api.getPools(request),
           RxOp.map(RD.success),
+          RxOp.startWith(RD.pending),
           RxOp.catchError((e: Error) => Rx.of(RD.failure(e)))
         )
-      )
+      ),
+      liveData.chain((details) =>
+        FP.pipe(
+          details,
+          A.map(({ asset }) =>
+            FP.pipe(
+              midgardDefailtApi$,
+              liveData.chain((api) =>
+                FP.pipe(
+                  // As midgard v2 is missing poolSlipAverage and swappingTxCount
+                  // field we have to get them from getSwapHistory
+                  api.getSwapHistory({
+                    pool: asset,
+                    interval: GetSwapHistoryIntervalEnum.Year,
+                    // Emulate ALL period of a pool life
+                    // And get data for last 10 years
+                    from: Date.now() - 360 * 3600 * 24 * 10,
+                    to: Date.now()
+                  }),
+                  RxOp.map(RD.success),
+                  liveData.map(({ meta }) => meta),
+                  RxOp.catchError(() => Rx.of(RD.failure(Error('Failed to load swaps history')))),
+                  RxOp.startWith(RD.pending)
+                )
+              ),
+              liveData.map(({ averageSlip, totalCount }) => ({
+                poolSlipAverage: averageSlip,
+                swappingTxCount: totalCount
+              })),
+              /**
+               * Set default value for all failed items to avoid falling into
+               * the error branch on the next step by liveData.sequenceArray
+               */
+              liveData.altOnError((): { poolSlipAverage: string; swappingTxCount: string } => ({
+                poolSlipAverage: '0',
+                swappingTxCount: '0'
+              }))
+            )
+          ),
+          liveData.sequenceArray,
+          liveData.map(
+            A.mapWithIndex((index, swapData) => ({
+              ...swapData,
+              // Here we can be sure in indexing 'cuz we have the same order
+              // as details 'cuz we build this array based on details themselves
+              ...details[index]
```

Comment:
_Just curious:_ What happened if the API will change the order of results?

Reply:
@veado nothing will happen. everything will work as expected.
what is happening here:
- we are going through all pool details' with simple `A.map`
- - for each pool we will request appropriate swaps history 
- - for all failed request we set mocked `RD.success` value. so we have `(RD.initial | RD.pending | RD.success)[]` (no `RD.failure`) to avoid failing to the error on the next step by sequencing with `liveData.sequenceArray`
- in other words we do Array.map `pool => shwapHistoryByPool` (order is the same).
- sequence `Observable<RD[]>` to the `Observable<[RD]>` with `liveData.sequenceArray` (it will just use `combineLatest`). this will not lead to any reorder
- as a result we have 1to1 match (by arrays' indexes) between `details` and `requested swaps history`